### PR TITLE
Add PHPUnit support for testing PHP

### PR DESCRIPTION
**Pull request by ankr**
**First draft implementation of PHPUnit for testing.**

# Installation
Using composer to install PHPUnit with dependencies.
```
composer intall
```

# Tests
So far I've only tested my implementation using the `node run` interface. Here are some of the tests I've been using. With a little more time and knowledge about the system I will create real test cases.

## Simple var test
Testing variables from user script is available in test method.
```bash
node run -l php \
-c '$a = "foo";' \ 
-f '$this->assertEquals("foo", $a);'
```

## Function test
Testing functions can be called and tested.
```bash
node run -l php -c '
function do_you_play_banjo($name) {
  return strtolower($name[0]) === "r" ? $name . " plays banjo" : $name . " does not play banjo";
}
' -f '
$this->assertEquals("ronnie plays banjo", do_you_play_banjo("ronnie"));
$this->assertEquals("mark does not play banjo", do_you_play_banjo("mark"));'
```

## Class test
Testing classes and mock objects can be created.
```bash
node run -l php -c '
class A {
  function f1() {
    $this->f2("foo");
  }
  function f2() {
  }
}
' -f '
$mock = $this->getMock("A", ["f2"]);
$mock->expects($this->once())
  ->method("f2")
  ->with("foo");
$mock->f1();'
```

# Successful test
If all expectations passes the output is `All test passed`

# Failed test
Example output of when we make the "Class test" fail.
```
PHPUnit 4.3.0 by Sebastian Bergmann.

F

Time: 21 ms, Memory: 3.00Mb

There was 1 failure:

1) KataTest::testSolution
Expectation failed for method name is equal to <string:f2> when invoked 1 time(s)
Parameter 0 for invocation A::f2('foo') does not match expected value.
Failed asserting that two strings are equal.
--- Expected
+++ Actual
@@ @@
-'bar'
+'foo'

FAILURES!
Tests: 1, Assertions: 0, Failures: 1.
```

### CODE REVIEW

File: lib/runners/php.js [hunk 1]
```diff
@@ -18,13 +18,29 @@ module.exports.run = function run(opts, cb)
...
+                opts.fixture,
+                config.snippets.php.classEnd,
+                config.snippets.php.execute
+            ].join('\n');
 
+            return {name: 'php', 'args': ['-r', code]};
+        },
+        sanitizeStdOut : function (output) {
+            var lines = output.split('\n');
+            var success = (lines[lines.length - 2] || '').slice(0, 2) === 'OK';
```
Comment (xcthulhu) [inline lib/runners/php.js]: Hi Codewars/codewars-runner,
A message from Matthew Wampler-Doty was left on IQ Elite for you. 
Click here a.iqelite.com/j/0JmRlo/f3r/?hg=J4GBUSzi6k to read your message 


The IQ Elite Team



---------------------------------------------------------------------------
This message was meant for reply@reply.github.com. If you do not want to continue receiving such mails from IQ Elite, please click on this link: a.iqelite.com/common/unsubscribe/?code=00d15143&email=reply@reply.github.com&hg=J4GBUSzi6k
IQ Elite is a service with main office: C/ Gran De Gràcia 15, 1a 1a, 08012 Barcelona, Spain.
---------------------------------------------------------------------------

Comment (xcthulhu) [inline lib/runners/php.js]: Hello,

Matthew  started using IQ Elite. He  wrote you a message.

To read the message before it expires, just click on this link: a.iqelite.com/entry/new_message/304c5d3a9b82edb35f6db34c2d3fda9e1f544b80bfb/320c612f75030113c58896d667b5182903c895f74747b907491a88260b6745f9058517ccb67b689665f60ccbebd5859a890908967ccbeccc5b63c587b897b8274c320c612f75044d3e22edb35af44b80bfb/?hg=vtlqCpN9wQ

Read your message before it expires!            

Regards,
IQ Elite Team



---------------------------------------------------------------------------
This message was meant for reply@reply.github.com. If you do not want to continue receiving such mails from IQ Elite, please click on this link: a.iqelite.com/common/unsubscribe/?code=00d15143&email=reply@reply.github.com&hg=vtlqCpN9wQ
IQ Elite is a service with main office: C/ Gran De Gràcia 15, 1a 1a, 08012 Barcelona, Spain.
---------------------------------------------------------------------------

Comment (xcthulhu) [inline lib/runners/php.js]: Hello,

Your friend Matthew sent you a message 36h days ago, which is still unread.

Within 24 hours the message will be deleted.

If you are curious to what Matthew wrote you click here: a.iqelite.com/entry/new_message/304c5d3a9b82edb35f6db34c2d3fda9e1f544b80bfb/320c612f75030113c58896d667b5182903c895f74747b907491a88260b6745f9058517ccb67b689665f60ccbebd5859a890908967ccbeccc5b63c587b897b8274c320c612f75044d3e22edb35af44b80bfb/?hg=caTXHKqtAf            

Regards,
IQ Elite Team



---------------------------------------------------------------------------
This message was meant for reply@reply.github.com. If you do not want to continue receiving such mails from IQ Elite, please click on this link: a.iqelite.com/common/unsubscribe/?code=00d15143&email=reply@reply.github.com&hg=caTXHKqtAf
IQ Elite is a service with main office: C/ Gran De Gràcia 15, 1a 1a, 08012 Barcelona, Spain.
---------------------------------------------------------------------------

**Comment by jhoffner** [comment]
@ankr I have merged your changes in but the integration is incomplete. We still need specs and examples. The project has changed a good amount so let me know if you have the time and need any help. The readme has a good amount of info.

**Pull request closed without merging**